File: tests/fixtures/java/control-flow.java
package fixtures.control;

import java.util.List;

class ControlFlow {
    interface Callback {
        void run();
    }

    interface Resource extends AutoCloseable {}

    Callback make() {
        return null;
    }

    Resource open() {
        return null;
    }

    boolean ready() {
        return true;
    }

    void work() {}

    void control(List<Callback> callbacks, Object value) {
        for (Callback loop = make(); ready(); loop = make()) {
            loop.run();
        }
        loop.run();

        for (Callback item : callbacks) {
            item.run();
        }
        item.run();

        try (Resource resource = open()) {
            resource.close();
        }
        resource.close();

        try {
            work();
        } catch (RuntimeException error) {
            error.printStackTrace();
        }
        error.printStackTrace();

        if (value instanceof Callback callback) {
            callback.run();
        }
        callback.run();

        if (value instanceof Pair(Callback first, Callback second)) {
            first.run();
            second.run();
        }
        first.run();

        switch (value) {
            case Callback selected -> selected.run();
            default -> {}
        }
        selected.run();

        while (value instanceof Callback repeated) {
            repeated.run();
            break;
        }
        repeated.run();
    }
}

File: tests/fixtures/csharp/control-flow.cs
using System;
using System.Collections.Generic;

namespace Fixtures.Control;

delegate void Callback();

sealed class Resource : IDisposable
{
    public void Dispose() {}
}

sealed class Failure : Exception
{
    public bool IsExpected() => true;
    public void Report() {}
}

class ControlFlow
{
    Callback Make() => null!;
    Resource Open() => new();
    bool Ready() => true;
    void Work() {}

    void Control(IEnumerable<Callback> callbacks, object value)
    {
        for (Callback loop = Make(); Ready(); loop = Make())
        {
            loop();
        }
        loop();

        foreach (Callback item in callbacks)
        {
            item();
        }
        item();

        using (Resource resource = Open())
        {
            resource.Dispose();
        }
        resource.Dispose();

        try
        {
            Work();
        }
        catch (Failure error) when (error.IsExpected())
        {
            error.Report();
        }
        error.Report();

        if (value is Callback selected)
        {
            selected();
        }
        selected();

        switch (value)
        {
            case Callback matched:
                matched();
                break;
        }
        matched();

        while (value is Callback repeated)
        {
            repeated();
            break;
        }
        repeated();
    }
}
